src/library/crate/cratestorage.h

```diff
@@ -179,21 +176,17 @@ class CrateTrackSelectResult: public FwdSqlQuerySelectResult {
     CrateTrackQueryFields m_queryFields;
 };
 
-class CrateStorage: public SqlStorage {
+class CrateStorage: public virtual /*implements*/ SqlStorage {
```

Comment:
I use virtual inheritance as a default for interface inheritance, i.e. no implementation. A class should only have a single non-virtual parent. In this case it has non, i.e. no dedicated base class. SqlStorage is not a base class, but just a collection of pure virtual functions that define an interface. You may have a base class and still _implement_ SqlStorage.